modules/core/rapid_context.js

```diff
@@ -1,34 +1,73 @@
-import { geoExtent } from '../geo';
+import { geoExtent, geoExtentFromBounds } from '../geo';
 
 import toGeoJSON from '@mapbox/togeojson';
-
+import { dispatch as d3_dispatch } from 'd3-dispatch'; 
+import { utilRebind } from '../util'
 
 export function coreRapidContext(context) {
     var rapidContext = {};
     rapidContext.version = '1.0.1';
+    var _isRect = undefined; 
+    var dispatch = d3_dispatch('task_extent_set');
```

Comment:
I see you there, wondering why I need to use pub-sub to display a static option correctly. 

The simple answer is- the gpx task URL is read sometime AFTER the first few renders are called for the layer sidebar. That means at time t=0, no task gpx exists, so we render the control with no grid options. 

The 'task_extent_set' event fires once the extent info has been read from the URL and is available for the sidebar to re-render.

Reply:
If the task extent is not rectangular, we don't bother re-rendering the sidebar. If it IS rectangular, then we re-render the sidebar with the grid controls in place.